curp/src/server/raw_curp/mod.rs

```diff
@@ -310,6 +313,16 @@ impl<C: 'static + Command, RC: RoleChange + 'static> RawCurp<C, RC> {
                 Err(ConfChangeError::new_propose(ProposeError::NotLeader)),
             );
         }
+        if let Some(change) = conf_change.changes().first() {
+            if let ConfChangeType::Promote = change.change_type() {
+                let learner_index = self.lst.get_match_index(change.node_id);
+                let leader_index = self.log.read().last_log_index();
+                if leader_index.overflow_sub(learner_index) > MAX_PROMOTE_GAP {
+                    return (info, Err(ConfChangeError::LearnerNotCatchUp(())));
+                }
+            }
+        }
```

Comment:
Should these be checked in `check_new_config`?

Reply:
It seems that the conf change was inserted into logs before we checked for failures. Is there a problem with this?

Reply:
Fixed